File: apple/Sources/ReactNativeWatchOS/Modifiers/ToggleStyle.swift
import SwiftUI

/// SwiftUI `.toggleStyle(_:)`. Maps the `style` string to a toggle style.
/// Unknown values fall back to `.automatic`.
enum RNWToggleStyleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("toggleStyle") { view, params, _ in
            switch params.string("style") {
            case "button":
                return AnyView(view.toggleStyle(.button))
            case "switch":
                return AnyView(view.toggleStyle(.switch))
            default:
                return AnyView(view.toggleStyle(.automatic))
            }
        }
    }
}

File: src/modifiers/toggleStyle.ts
import { createModifier } from './createModifier';

export type ToggleStyle = 'automatic' | 'button' | 'switch';

/// SwiftUI `.toggleStyle(_:)`. Sets the visual style applied to `Toggle`s
/// within the view. `'automatic'` defers to the platform default.
export function toggleStyle(style: ToggleStyle = 'automatic') {
  return createModifier('toggleStyle', { style });
}
